From 37e529642a859dd1f6ce05eaa9dbecba21a3df09 Mon Sep 17 00:00:00 2001
From: Idan Frimark <40820488+FrimIdan@users.noreply.github.com>
Date: Wed, 23 Feb 2022 16:19:46 +0200
Subject: fix an issue where the mounted external CA secret was ignored
 (#37462)

* fix an issue where the mounted external CA secret was ignored

* review
---
 pilot/pkg/bootstrap/istio_ca.go | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/pilot/pkg/bootstrap/istio_ca.go b/pilot/pkg/bootstrap/istio_ca.go
index 9bc4f606ae..023e6c4e92 100644
--- a/pilot/pkg/bootstrap/istio_ca.go
+++ b/pilot/pkg/bootstrap/istio_ca.go
@@ -477,10 +477,19 @@ func (s *Server) createIstioRA(client kubelib.Client,
 	caCertFile := path.Join(ra.DefaultExtCACertDir, constants.CACertNamespaceConfigMapDataName)
 	certSignerDomain := opts.CertSignerDomain
 	_, err := os.Stat(caCertFile)
-	if err != nil && certSignerDomain == "" {
-		caCertFile = defaultCACertPath
-	} else {
-		caCertFile = ""
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, fmt.Errorf("failed to get file info: %v", err)
+		}
+
+		// File does not exist.
+		if certSignerDomain == "" {
+			log.Infof("CA cert file %q not found, using %q.", caCertFile, defaultCACertPath)
+			caCertFile = defaultCACertPath
+		} else {
+			log.Infof("CA cert file %q not found - ignoring.", caCertFile)
+			caCertFile = ""
+		}
 	}
 	raOpts := &ra.IstioRAOptions{
 		ExternalCAType:   opts.ExternalCAType,
-- 
2.35.3

